### PR TITLE
Use fluxible@0.4.x; Convert to ES6; Use eslint and babel-eslint

### PR DESCRIPTION
**Pull request by mridgway**

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+  parser: "babel-eslint"
+  env:
+    node: true
+  rules:
+    strict: 0
+    indent: [2, 4]
+    quotes: [2, 'single']
+    no-unused-vars: 0 // see https://github.com/babel/babel-eslint/issues/21
```
Comment (redonkulus) [inline .eslintrc]: Does this cover all our jscs rules? Seems like we are missing a bunch.

File: app/templates/_package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "private": true,
   "scripts": {
     "dev": "grunt",
-    "lint": "grunt jshint"
+    "lint": "eslint ./*.js ./**/*.js"
```
Comment (redonkulus) [inline app/templates/_package.json]: This doesn't lint node_modules right?

**Comment by redonkulus** [comment]
Overall changes look good, just some minor comments.

File: app/templates/_package.json (hunk 2 above)
Comment (mridgway) [inline app/templates/_package.json]: No, eslint ignores node_modules by default.

File: .eslintrc (hunk 1 above)
Comment (mridgway) [inline .eslintrc]: Not sure, the list is different from jshint's so would take some time to match up.

Comment (redonkulus) [inline .eslintrc]: Does eslint support jscs style formatting too?

**Comment by redonkulus** [comment]
:shipit:

File: .eslintrc (hunk 1 above)
Comment (hzoo) [inline .eslintrc]: You can use https://github.com/yannickcr/eslint-plugin-react to fix any jsx/react issues.

**Pull request merged**